lib/SILOptimizer/Mandatory/MandatoryCombiner.cpp

```diff
@@ -143,20 +143,25 @@ struct MandatoryCombiner : SILFunctionTransform {
           madeChange |= result.first || result.second;
         }
 
+        // Whether the current instruction was deleted or not, because we
+        // already moved the iterator back a step, we need to move it forward a
+        // step.  If the current instruction was in fact deleted, moving the
+        // iterator forward will not advance it to the current instruction, it
+        // having been deleted, but will instead advance it to the next
+        // instruction that remains.
+        ii = next_or_default(ii, /*end=*/block.end(),
+                             /*defaultIter=*/block.begin());
         if (deleted) {
-          // The deletion succeeded.  The iterator has already been moved back
-          // a step so as to avoid invalidation.  Advancing it one step will
-          // not advance it to the current instruction since that has been
-          // deleted.  Instead, it will advance it to the next *remaining*
-          // instruction after the previous instruction as desired.
-          ii = next_or_default(ii, /*end*/block.end(), /*defaultIter*/block.begin());
+          // The current instruction has been deleted.  The iterator has already
```

Comment:
Can you move any changes that are not directly related to adding this to the pass pipeline into a different PR (or at least a different commit)?

Reply:
Yup!  Will split into separate commit.

Reply:
https://github.com/apple/swift/pull/26924/commits/e4cee7fdfaac4051b8b75f7f042f93d20f22bb1e